src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java

```diff
@@ -62,4 +66,25 @@ public FinancialTransactionDTO findById(@Min(1) @NotNull Long id) {
                 .orElseThrow(() -> new AppRuntimeException(ErrorCode.FT01,
                         String.format("Financial transaction with id: %d not found", id)));
     }
+
+    @Override
+    @Transactional
+    public FinancialTransactionDTO updateTransaction(
+            @Min(1) @NotNull Long id,
+            @Valid UpdateFinancialTransactionDTO updateTransactionDTO){
```

Comment:
On Controller you also got @NotNull for this argument. Why there is @NotNull here?